### PR TITLE
Directory structure maker script addition.

### PR DESCRIPTION
**Pull request by slahirucd7**
### Purpose
Adds directory structure maker script to the repo.
It creates below directory structure.

```
choreo-connect-1.1.0
├── ADAPTER
│   ├── $HOME
│   │   ├── LICENSE.txt
│   │   ├── adapter-linux-amd64
│   │   ├── adapter-linux-arm64
│   │   ├── check_health.sh
│   │   ├── conf
│   │   │   ├── config.toml
│   │   │   └── log_config.toml
│   │   └── security
│   │       ├── keystore
│   │       │   ├── mg.key
│   │       │   └── mg.pem
│   │       └── truststore
│   │           ├── consul
│   │           │   ├── consul-agent-ca.pem
│   │           │   ├── local-dc-client-consul-0-key.pem
│   │           │   └── local-dc-client-consul-0.pem
│   │           ├── controlplane.pem
│   │           └── mg.pem
│   ├── Dockerfile
│   ├── Dockerfile.ubuntu
│   └── bin
│       ├── grpc_health_probe-linux-amd64
│       └── grpc_health_probe-linux-arm64
├── ENFORCER
│   ├── $HOME
│   │   ├── LICENSE.txt
│   │   ├── check_health.sh
│   │   ├── conf
│   │   │   └── log4j2.properties
│   │   ├── lib
│   │   │   ├── dropins
│   │   │   ├── json-3.0.0.wso2v1.jar
│   │   │   ├── log4j-api-2.17.1.jar
│   │   │   ├── log4j-core-2.17.1.jar
│   │   │   ├── log4j-jcl-2.17.1.jar
│   │   │   ├── log4j-slf4j-impl-2.17.1.jar
│   │   │   ├── org.wso2.choreo.connect.enforcer-1.1.0-javadoc.jar
│   │   │   ├── org.wso2.choreo.connect.enforcer-1.1.0-sources.jar
│   │   │   └── org.wso2.choreo.connect.enforcer-1.1.0.jar
│   │   └── security
│   │       ├── keystore
│   │       │   ├── mg.key
│   │       │   └── mg.pem
│   │       └── truststore
│   │           ├── mg.pem
│   │           └── wso2carbon.pem
│   ├── Dockerfile
│   ├── Dockerfile.ubuntu
│   └── bin
│       ├── grpc_health_probe-linux-amd64
│       └── grpc_health_probe-linux-arm64
├── ROUTER
│   ├── $HOME
│   │   ├── interceptor
│   │   │   └── lib
│   │   │       ├── base64.lua
│   │   │       ├── consts.lua
│   │   │       ├── encoders.lua
│   │   │       ├── interceptor.lua
│   │   │       ├── json.lua
│   │   │       └── utils.lua
│   │   └── wasm
│   │       └── websocket
│   │           └── mgw-websocket.wasm
│   ├── Dockerfile
│   ├── Dockerfile.ubuntu
│   ├── LICENSE.txt
│   └── etc
│       ├── envoy
│       │   └── envoy.yaml
│       └── ssl
│           └── certs
│               └── ca-certificates.crt
├── bin
└── updates
    └── product.txt

```

---

**Execution:**

Screenshot 2022-04-19 at 10 59 06

### Issues
N/A

Fixes https://github.com/wso2/product-microgateway/issues/2883

### Automation tests
 - Unit tests added: No
 - Integration tests added: No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: dev-scripts/directory-structure-maker/directory-structure-maker.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+│           └── certs
+│               └── ca-certificates.crt
+├── bin
+└── updates
+    └── product.txt
+```
+
+## Prerequisites
+1. Before running this script you need to run Choreo Connect docker-compose file up and running in the background.
+2. You should execute the script with sudo permissions. (Otherwise there will be errors when executing `docker cp` command.)
```
Comment (pubudu538) [inline dev-scripts/directory-structure-maker/directory-structure-maker.md:93]: We don't need sudo here. You can configure docker to run in non root user - https://docs.docker.com/engine/install/linux-postinstall/#manage-docker-as-a-non-root-user

File: dev-scripts/directory-structure-maker/directory-structure-maker.sh [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+cp $CC_REPO_PATH/enforcer-parent/enforcer/src/main/resources/Dockerfile         ./ENFORCER
+cp $CC_REPO_PATH/enforcer-parent/enforcer/src/main/resources/Dockerfile.ubuntu  ./ENFORCER
+cp $CC_REPO_PATH/router/src/main/resources/Dockerfile                           ./ROUTER
+cp $CC_REPO_PATH/router/src/main/resources/Dockerfile.ubuntu                    ./ROUTER
+
+# Creates update directory txt
+touch ./updates/product.txt
+echo "choreo-connect-"$CC_VERSION >> ./updates/product.txt
+
+echo "Choreo Connect folder structure prepared successfully!"
```
Comment (pubudu538) [inline dev-scripts/directory-structure-maker/directory-structure-maker.sh:95]: I think we use remove-container script in other places. Check for usages.

Comment (slahirucd7) [reply dev-scripts/directory-structure-maker/directory-structure-maker.sh:95]: Fixed with: https://github.com/wso2/product-microgateway/pull/2875/commits/a876184e5ff849920343eef2801e399fa2aa211c

File: pom.xml [hunk 3]
```diff
@@ -160,7 +160,7 @@
                                     <fileset dir="resources/docker-compose" includes="**/docker-compose.yaml" />
                                     <fileset dir="resources/k8s-artifacts" includes="**/*deployment.yaml" />
                                 </replaceregexp>
-                                <replaceregexp file="remove-containers.sh" match="micro_gw_version=.*" replace="micro_gw_version=${project.version}" byline="true" />
+                                <replaceregexp file="dev-scripts/remove-containers.sh" match="micro_gw_version=.*" replace="micro_gw_version=${project.version}" byline="true" />
```
Comment (renuka-fernando) [inline pom.xml:163]:
```suggestion
                                <replaceregexp file="dev-scripts/remove-containers.sh" match="cc_version=.*" replace="cc_version=${project.version}" byline="true" />
```

**Review by pubudu538 — APPROVED** [review]

File: dev-scripts/directory-structure-maker/directory-structure-maker.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+│   │       ├── keystore
+│   │       │   ├── mg.key
+│   │       │   └── mg.pem
+│   │       └── truststore
+│   │           ├── consul
+│   │           │   ├── consul-agent-ca.pem
+│   │           │   ├── local-dc-client-consul-0-key.pem
+│   │           │   └── local-dc-client-consul-0.pem
+│   │           ├── controlplane.pem
+│   │           ├── global-adapter.pem
```
Comment (VirajSalaka) [inline dev-scripts/directory-structure-maker/directory-structure-maker.md:29]: Do we need global adapter pem file to be included?

Comment (slahirucd7) [reply dev-scripts/directory-structure-maker/directory-structure-maker.md:29]: Checked in the pack prepared for the 1.0.0 onboarding. In there also `global-adapter.pem` file is there.

Comment (pubudu538) [reply dev-scripts/directory-structure-maker/directory-structure-maker.md:29]: Let's remove that

**Review by pubudu538 — APPROVED** [review]

**Pull request merged**